Sming/Core/Network/Http/HttpServerConnection.cpp

```diff
@@ -103,34 +103,46 @@ int HttpServerConnection::onHeadersComplete(const HttpHeaders& headers)
 		error = resource->onHeadersComplete(*this, request, response);
 	}
 
-	if(!error && request.method == HTTP_HEAD) {
-		error = 1;
-	}
-
-	if(bodyParsers != nullptr && request.headers.contains(HTTP_HEADER_CONTENT_TYPE)) {
-		String contentType = request.headers[HTTP_HEADER_CONTENT_TYPE];
-		int endPos = contentType.indexOf(';');
-		if(endPos != -1) {
-			contentType = contentType.substring(0, endPos);
+	if(!error) {
```

Comment:
@aemseemann Refactor this block and try to avoid many levels of nested ifs.

Reply:
I'd also prefer to see `if (endPos >= 0)` as its more robust

Reply:
Done.

Hope that it's ok to skip all the body parser setup in case `error != 0`. Although this change is not strictly necessary for this PR, it seemed futile to prepare for a body that will never come.